src/include/machnet_engine.h

```diff
@@ -874,7 +962,73 @@ class MachnetEngine {
         if (listeners_.find(local_ipv4_addr) != listeners_.end()) {
           // We have a listener on this port.
           const auto &listeners_on_ip = listeners_[local_ipv4_addr];
+          const int qid =
+              shared_state_->GetQueueId(local_ipv4_addr, local_udp_port);
           if (listeners_on_ip.find(local_udp_port) == listeners_on_ip.end()) {
+            if (qid > -1) {
+              auto *response = CHECK_NOTNULL(packet_pool_->PacketAlloc());
+              const size_t hdr_length =
+                  (sizeof(Ethernet) + sizeof(Ipv4) + sizeof(Udp) +
+                   sizeof(net::MachnetPktHdr));
+              const uint32_t pkt_len =
+                  hdr_length +
+                  46;  // I am assuming the rss key size to 40 bytes.
+              Ethernet *response_eh = response->append<Ethernet *>(pkt_len);
+
+              // THIS IS A BIG PROBLEM!
+              LOG(WARNING) << pkt->length() << " XXXX " << pkt_len;
+              if (!response_eh) LOG(WARNING) << "could not append";
+              // Size is different that just packet->length;
+              response_eh->dst_addr = eh->src_addr;
+              response_eh->src_addr = pmd_port_->GetL2Addr();
+              response_eh->eth_type = be16_t(Ethernet::kIpv4);
+              response->set_l2_len(sizeof(*response_eh));
+              Ipv4 *response_ipv4h = reinterpret_cast<Ipv4 *>(response_eh + 1);
+              response_ipv4h->version_ihl = 0x45;
+              response_ipv4h->type_of_service = 0;
+              response_ipv4h->packet_id = be16_t(0x1513);
+              response_ipv4h->fragment_offset = be16_t(0);
+              response_ipv4h->next_proto_id = Ipv4::Proto::kUdp;
+              response_ipv4h->time_to_live = 64;
+              // response_ipv4h->total_length = be16_t(pkt->length() -
+              // sizeof(Ethernet));
+              response_ipv4h->total_length = be16_t(pkt_len - sizeof(Ethernet));
+              response_ipv4h->src_addr = ipv4h->dst_addr;
+              response_ipv4h->dst_addr = ipv4h->src_addr;
+              response_ipv4h->hdr_checksum = 0;
+              response->set_l3_len(sizeof(*response_ipv4h));
+              response->offload_ipv4_csum();  // I should calculate the checksum
```

Comment:
Please remove the comment. Also add a line break after this for readability.

Similarly all the comments below for debugging/leftover code (e.g., 1004).